dm/pkg/ha/bound.go

```diff
@@ -128,31 +128,35 @@ func ReplaceSourceBound(cli *clientv3.Client, source, oldWorker, newWorker strin
 
 // GetSourceBound gets the source bound relationship for the specified DM-worker.
 // if the bound relationship for the worker name not exist, return with `err == nil`.
+// if source
 // if the worker name is "", it will return all bound relationships as a map{worker-name: bound}.
 // if the worker name is given, it will return a map{worker-name: bound} whose length is 1.
-func GetSourceBound(cli *clientv3.Client, worker string) (map[string]SourceBound, int64, error) {
+func GetSourceBound(cli *clientv3.Client, worker, source string) (map[string]map[string]SourceBound, int64, error) {
 	ctx, cancel := context.WithTimeout(cli.Ctx(), etcdutil.DefaultRequestTimeout)
 	defer cancel()
 
 	var (
-		sbm  = make(map[string]SourceBound)
+		sbm  = make(map[string]map[string]SourceBound)
 		resp *clientv3.GetResponse
 		err  error
 	)
 	failpoint.Inject("FailToGetSourceCfg", func() {
 		failpoint.Return(sbm, 0, context.DeadlineExceeded)
 	})
-	if worker != "" {
-		resp, err = cli.Get(ctx, common.UpstreamBoundWorkerKeyAdapter.Encode(worker))
-	} else {
+	switch {
+	case source != "":
+		resp, err = cli.Get(ctx, common.UpstreamBoundWorkerKeyAdapter.Encode(worker, source))
+	case worker != "":
+		resp, err = cli.Get(ctx, common.UpstreamBoundWorkerKeyAdapter.Encode(worker), clientv3.WithPrefix())
+	default:
 		resp, err = cli.Get(ctx, common.UpstreamBoundWorkerKeyAdapter.Path(), clientv3.WithPrefix())
```

Comment:
seems should use `UpstreamBoundWorkerKeyAdapterV1` ? for existing cluster? do we have some ut for this case ?

Reply:
We use this new key in all the current unit tests.
We will fix the upgrade problem for the existing cluster in next PR. This PR is also a feature branch now.

Reply:
ooo i see